### PR TITLE
Managed SDK UI (behind feature flag)

### PR DESCRIPTION
**Pull request by ivanporty**
PR contains purely UI for choosing between Managed/Custom SDK. Feature flag is used to show managed SDK UI (set `feature.managed.sdk = true` in `config.properties` file to enable).

### Managed SDK UI:
managedsdk-1

managedsdk-custom

### Default UI (managed SDK feature disabled).

Settings:

old-sdk-settings

Local Dev Server settings (re-used panel):

old-sdk-applocal

### CODE REVIEW

**Comment by etanshaul** [comment]
nit: instead of just "managed" what about something more descriptive (to indicate that its automatically managed (installed) by the plugin (like you have for custom)? 

Something like "Managed (Installation managed by the plugin)" or something along those lines?

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java [hunk 1]
```diff
@@ -45,6 +54,11 @@
   private JTextPane warningMessage;
   private JPanel cloudSdkPanel;
   private JLabel warningIcon;
+  private JRadioButton managedRadioButton;
+  private JRadioButton customRadioButton;
+  private JCheckBox enableAutomaticUpdatesCheckbox;
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java]: what about explicitly setting its default check/uncheck value and having a member boolean here containing that default? that way its easy to see what the default is and to change it later if we wanted.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java [hunk 2]
```diff
@@ -45,6 +54,11 @@
   private JTextPane warningMessage;
   private JPanel cloudSdkPanel;
   private JLabel warningIcon;
+  private JRadioButton managedRadioButton;
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java]: what determines if "managed" or "custom" will be selected by default? is this for a different PR?

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java (hunk 1 above)
Comment (ivanporty) [reply google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java]: Sure, good idea.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java (hunk 2 above)
Comment (ivanporty) [reply google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java]: Yes, for the next PR - I'll add settings that will check and select proper value.

**Comment by ivanporty** [comment]
I agree, we can have better name... `Managed (installed by plugin)`? I'm open for suggestions here, can't think of really clean label so far.

**Comment by etanshaul** [comment]
yeah I can't really think of anything either. I think that will be ok for now (`Managed (installed by plugin)`).

**Comment by etanshaul** [comment]
I just thought to glance at the JB UI guide that @patflynn linked to. Specifically the rules around text and capitalization:

https://jetbrains.github.io/ui/text/capitalization/

They suggest to use sentences for radio buttons / checkboxes and have first-word-capitalized-only capitalization for these.

Not saying necessarily we should do this. But perhaps it would solve the other problem (of unclear wording).

**Comment by ivanporty** [comment]
Probably right. Let me change that too.

**Comment by ivanporty** [comment]
sdk-ui-2-managed

sdk-ui-custom-2

**Comment by etanshaul** [comment]
I think the guide was suggesting not just about the capitalization, but also making them sentences.

**Comment by etanshaul** [comment]
(only if you think it would help though)

**Comment by ivanporty** [comment]
`Managed and installed by plugin` and
`Custom local installation`?

**Comment by etanshaul** [comment]
hmm. if we go that way we might as well go all the way and make it a proper sentence. i.e.:

`Allow the plugin to manage and install the Cloud SDK`
`Use a custom local installation`

**Comment by ivanporty** [comment]
Sounds good to me.

**Comment by ivanporty** [comment]
sdk-ui-sentence

**Comment by etanshaul** [comment]
nit: "Automatic updates" as a sentence for consistency?

**Comment by etanshaul** [comment]
for example "Automatically update the SDK"

**Comment by ivanporty** [comment]
sdk-ui-final

**Pull request merged**